### ISSUE TITLE
pylance doesn't recognize installed dependency

### ISSUE THREAD

**Issue by mironnn** [microsoft/pylance-release#4545]
Type: <b>Bug</b>

# Behaviour
## Expected vs. Actual
Import error for the installed dependecy in virtual environment. Interpreter selected.
XXX

Import error.
## Steps to reproduce:

1. Create venv by python3.10 -m venv venv
2. Select interpreter path
3. install dependency in terminal
4 try to impor tin in code

# Diagnostic data

-   Python version (& distribution if applicable, e.g. Anaconda): 3.10.11
-   Type of virtual environment used (e.g. conda, venv, virtualenv, etc.): Venv
-   Value of the `python.languageServer` setting: Default

<details>

<summary>Output for <code>Python</code> in the <code>Output</code> panel (<code>View</code>→<code>Output</code>, change the drop-down the upper-right of the <code>Output</code> panel to <code>Python</code>)
</summary>

<p>

```
XXX
```

</p>
</details>

<details>

<summary>User Settings</summary>

<p>

```

languageServer: "Pylance"

linting
• flake8Args: "<placeholder>"
• flake8Enabled: true
• mypyEnabled: true
• pycodestyleArgs: "<placeholder>"
• pydocstyleEnabled: true

formatting
• provider: "black"

```

</p>
</details>

Extension version: 2023.10.1
VS Code version: Code 1.79.2 (695af097c7bd098fbf017ce3ac85e09bbc5dda06, 2023-06-14T08:58:33.551Z)
OS version: Darwin arm64 22.5.0
Modes:
Remote OS version: Linux x64 5.4.0-148-generic

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 Pro (10 x 24)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|32.00GB (2.27GB free)|
|Process Argv|--crash-reporter-id c9b543f4-aa3e-4145-a03c-a8d5a1362fbb|
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|SSH: 200|
|OS|Linux x64 5.4.0-148-generic|
|CPUs|AMD EPYC 7313 16-Core Processor (64 x 1498)|
|Memory (System)|1007.82GB (964.38GB free)|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263cf:30335440
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
pythonvs932:30410667
py29gd2263cf:30773604
vsclangdf:30486550
c4g48928:30535728
dsvsc012:30540252
pynewext54:30695312
azure-dev_surveyone:30548225
3biah626:30602489
pyind779:30671433
89544117:30613380
pythonsymbol12:30671437
a9j8j154:30646983
showlangstatbar:30737416
vsctsb:30748421
pythonms35:30701012
03d35959:30757346
pythonfmttext:30731395
pythoncmv:30756943
fixshowwlkth:30771522
pythongtdpath:30769146
i26e3531:30769768
gsof1:30774496
dh2dc718:30776458
pythonidxptcf:30772540
pythondjangots:30772535
pythonnocebcf:30776496

```

</details>

**Comment by StellaHuang95** [comment]
Hi there, please provide a minimal self-contained code sample that demonstrates the issue. Could you also provide us some logs as described in [the troubleshooting guide](https://github.com/microsoft/pylance-release/blob/main/TROUBLESHOOTING.md#filing-an-issue)?

**Comment by mironnn** [comment]
@StellaHuang95 yes, sure.

Log from Python output

```
2023-06-27 19:16:18.040 [info] Experiment 'pythonPromptNewFormatterExt' is active
2023-06-27 19:16:18.040 [info] Experiment 'pythonPromptNewToolsExt' is active
2023-06-27 19:16:18.040 [info] Experiment 'pythonSurveyNotification' is active
2023-06-27 19:16:18.040 [info] Test server listening.
2023-06-27 19:16:18.040 [info] > conda info --json
2023-06-27 19:16:18.040 [info] > ~/miniconda3/bin/conda info --json
2023-06-27 19:16:18.040 [error] Failed to initialize source map support in extension
2023-06-27 19:16:18.040 [info] Found: /bin/python --> /bin/python3.8
2023-06-27 19:16:18.040 [info] Found: /bin/python2 --> /bin/python2.7
2023-06-27 19:16:18.040 [info] Found: /bin/python2.7 --> /bin/python2.7
2023-06-27 19:16:18.040 [info] Found: /bin/python3 --> /bin/python3.8
2023-06-27 19:16:18.040 [info] Found: /bin/python3.10 --> /bin/python3.10
2023-06-27 19:16:18.040 [info] Found: /bin/python3.11 --> /bin/python3.11
2023-06-27 19:16:18.040 [info] Found: /bin/python3.7 --> /bin/python3.7
2023-06-27 19:16:18.040 [info] Found: /bin/python3.8 --> /bin/python3.8
2023-06-27 19:16:18.076 [info] > ~/.local/share/virtualenvs/nsions/ms-python.python-2023.10.1/pythonFiles/interpreterInfo.py
2023-06-27 19:16:18.081 [info] Found: /usr/bin/python --> /usr/bin/python3.8
2023-06-27 19:16:18.081 [info] Found: /usr/bin/python2 --> /usr/bin/python2.7
2023-06-27 19:16:18.081 [info] Found: /usr/bin/python2.7 --> /usr/bin/python2.7
2023-06-27 19:16:18.081 [info] Found: /usr/bin/python3 --> /usr/bin/python3.8
2023-06-27 19:16:18.081 [info] Found: /usr/bin/python3.10 --> /usr/bin/python3.10
2023-06-27 19:16:18.081 [info] Found: /usr/bin/python3.11 --> /usr/bin/python3.11
2023-06-27 19:16:18.081 [info] Found: /usr/bin/python3.7 --> /usr/bin/python3.7
2023-06-27 19:16:18.081 [info] Found: /usr/bin/python3.8 --> /usr/bin/python3.8
2023-06-27 19:16:18.082 [info] Found: /usr/include/python3.7 --> /usr/include/python3.7m
2023-06-27 19:16:18.087 [info] Found: /usr/local/bin/python3.11 --> /usr/local/bin/python3.11
2023-06-27 19:16:18.119 [info] > /bin/python -I ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/get_output_via_markers.py ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/interpreterInfo.py
2023-06-27 19:16:18.296 [info] > /bin/python2 -I ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/get_output_via_markers.py ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/interpreterInfo.py
2023-06-27 19:16:18.299 [warning] Error: Command failed: /bin/python2 -I /home//.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/get_output_via_markers.py /home/user/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/interpreterInfo.py
Unknown option: -I
usage: /bin/python2 [option] ... [-c cmd | -m mod | file | -] [arg] ...
Try `python -h' for more information.

    at ChildProcess.exithandler (node:child_process:400:12)
    at ChildProcess.emit (node:events:513:28)
    at maybeClose (node:internal/child_process:1093:16)
    at Socket.<anonymous> (node:internal/child_process:451:11)
    at Socket.emit (node:events:513:28)
    at Pipe.<anonymous> (node:net:757:14) {
  code: 2,
  killed: false,
  signal: null,
  cmd: '/bin/python2 -I /home/user/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/get_output_via_markers.py /home/user/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/interpreterInfo.py'
}
2023-06-27 19:16:18.299 [error] Support for Python 2.7 has been dropped by the Python extension so certain features may not work, upgrade to using Python 3.
2023-06-27 19:16:18.309 [info] > /bin/python2 ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/get_output_via_markers.py ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/interpreterInfo.py
2023-06-27 19:16:18.328 [info] > /bin/python3.10 -I ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/get_output_via_markers.py ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/interpreterInfo.py
2023-06-27 19:16:18.350 [info] > /bin/python3.11 -I ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/get_output_via_markers.py ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/interpreterInfo.py
2023-06-27 19:16:18.376 [info] > /bin/python3.7 -I ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/get_output_via_markers.py ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/interpreterInfo.py
2023-06-27 19:16:18.404 [info] > /usr/bin/python -I ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/get_output_via_markers.py ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/interpreterInfo.py
2023-06-27 19:16:18.430 [info] > /usr/bin/python2 -I ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/get_output_via_markers.py ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/interpreterInfo.py
2023-06-27 19:16:18.434 [warning] Error: Command failed: /usr/bin/python2 -I /home/user/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/get_output_via_markers.py /home/user/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/interpreterInfo.py
Unknown option: -I
usage: /usr/bin/python2 [option] ... [-c cmd | -m mod | file | -] [arg] ...
Try `python -h' for more information.

    at ChildProcess.exithandler (node:child_process:400:12)
    at ChildProcess.emit (node:events:513:28)
    at maybeClose (node:internal/child_process:1093:16)
    at Socket.<anonymous> (node:internal/child_process:451:11)
    at Socket.emit (node:events:513:28)
    at Pipe.<anonymous> (node:net:757:14) {
  code: 2,
  killed: false,
  signal: null,
  cmd: '/usr/bin/python2 -I /home/user/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/get_output_via_markers.py /home/user/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/interpreterInfo.py'
}
2023-06-27 19:16:18.434 [error] Support for Python 2.7 has been dropped by the Python extension so certain features may not work, upgrade to using Python 3.
2023-06-27 19:16:18.454 [info] > /usr/bin/python2 ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/get_output_via_markers.py ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/interpreterInfo.py
2023-06-27 19:16:18.482 [info] > /usr/bin/python3.10 -I ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/get_output_via_markers.py ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/interpreterInfo.py
2023-06-27 19:16:18.485 [warning] Retry failed, storage update failed for key PYTHON_GLOBAL_STORAGE_KEYS
2023-06-27 19:16:18.485 [warning] Retry failed, storage update failed for key PYTHON_GLOBAL_STORAGE_KEYS
2023-06-27 19:16:18.485 [warning] Retry failed, storage update failed for key PYTHON_WORKSPACE_STORAGE_KEYS
2023-06-27 19:16:18.485 [warning] Retry failed, storage update failed for key PYTHON_WORKSPACE_STORAGE_KEYS
2023-06-27 19:16:18.485 [warning] Retry failed, storage update failed for key PYTHON_WORKSPACE_STORAGE_KEYS
2023-06-27 19:16:18.524 [info] > /usr/bin/python3.11 -I ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/get_output_via_markers.py ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/interpreterInfo.py
2023-06-27 19:16:18.577 [info] > /usr/bin/python3.7 -I ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/get_output_via_markers.py ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/interpreterInfo.py
2023-06-27 19:16:18.605 [info] > /usr/include/python3.7 -I ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/get_output_via_markers.py ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/interpreterInfo.py
2023-06-27 19:16:18.608 [error] Error: Command failed: /usr/include/python3.7 -I /home/user/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/get_output_via_markers.py /home/user/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/interpreterInfo.py
/bin/sh: 1: /usr/include/python3.7: Permission denied

    at ChildProcess.exithandler (node:child_process:400:12)
    at ChildProcess.emit (node:events:513:28)
    at maybeClose (node:internal/child_process:1093:16)
    at Socket.<anonymous> (node:internal/child_process:451:11)
    at Socket.emit (node:events:513:28)
    at Pipe.<anonymous> (node:net:757:14) {
  code: 126,
  killed: false,
  signal: null,
  cmd: '/usr/include/python3.7 -I /home/user/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/get_output_via_markers.py /home/user/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/interpreterInfo.py'
}
2023-06-27 19:16:18.628 [info] > /usr/local/bin/python3.11 -I ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/get_output_via_markers.py ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/interpreterInfo.py
2023-06-27 19:16:18.836 [info] > ~/miniconda3/bin/python -I ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/get_output_via_markers.py ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/interpreterInfo.py
2023-06-27 19:16:18.857 [info] > ~/miniconda3/envs/.venv/bin/python -I ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/get_output_via_markers.py ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/interpreterInfo.py
2023-06-27 19:16:20.636 [info] > /usr/include/python3.7 -I ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/get_output_via_markers.py ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/interpreterInfo.py
2023-06-27 19:16:20.638 [error] Error: Command failed: /usr/include/python3.7 -I /home/user/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/get_output_via_markers.py /home/user/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/interpreterInfo.py
/bin/sh: 1: /usr/include/python3.7: Permission denied

    at ChildProcess.exithandler (node:child_process:400:12)
    at ChildProcess.emit (node:events:513:28)
    at maybeClose (node:internal/child_process:1093:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5) {
  code: 126,
  killed: false,
  signal: null,
  cmd: '/usr/include/python3.7 -I /home/user/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/get_output_via_markers.py /home/user/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/interpreterInfo.py'
}
2023-06-27 19:16:20.752 [info] Python interpreter path: /bin/python3.11
2023-06-27 19:16:22.268 [info] LINTING: Skipping linting from Python extension, since Flake8 extension is installed and enabled.
2023-06-27 19:16:22.299 [info] > /bin/python3.11 ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/linter.py -m mypy --follow-imports=silent --ignore-missing-imports --show-column-numbers --no-pretty ./main.py
2023-06-27 19:16:22.299 [info] cwd: .
2023-06-27 19:16:22.320 [info] > /bin/python3.11 ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/linter.py -m pydocstyle ./main.py
2023-06-27 19:16:22.320 [info] cwd: .
2023-06-27 19:16:22.357 [info] Starting Pylance language server.
2023-06-27 19:16:22.415 [info] ##########Linting Output - pydocstyle##########

2023-06-27 19:16:22.415 [info] /home/user/temp/test-vscode/main.py:1 at module level:
        D100: Missing docstring in public module

2023-06-27 19:16:23.159 [info] ##########Linting Output - mypy##########

2023-06-27 19:16:23.159 [info] Success: no issues found in 1 source file

2023-06-27 19:16:51.516 [info] LINTING: Skipping linting from Python extension, since Flake8 extension is installed and enabled.
2023-06-27 19:16:51.539 [info] > /bin/python3.11 ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/linter.py -m mypy --follow-imports=silent --ignore-missing-imports --show-column-numbers --no-pretty ./main.py
2023-06-27 19:16:51.539 [info] cwd: .
2023-06-27 19:16:51.556 [info] > /bin/python3.11 ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/linter.py -m pydocstyle ./main.py
2023-06-27 19:16:51.556 [info] cwd: .
2023-06-27 19:16:51.633 [info] Discover tests for workspace name: test-vscode - uri: /home/user/temp/test-vscode/main.py
2023-06-27 19:16:51.645 [info] ##########Linting Output - mypy##########

2023-06-27 19:16:51.646 [info] main.py:1:9: error: invalid syntax  [syntax]
Found 1 error in 1 file (errors prevented further checking)

2023-06-27 19:16:51.656 [info] ##########Linting Output - pydocstyle##########

2023-06-27 19:16:51.656 [info] 
2023-06-27 19:16:53.899 [info] LINTING: Skipping linting from Python extension, since Flake8 extension is installed and enabled.
2023-06-27 19:16:53.933 [info] > /bin/python3.11 ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/linter.py -m mypy --follow-imports=silent --ignore-missing-imports --show-column-numbers --no-pretty ./main.py
2023-06-27 19:16:53.933 [info] cwd: .
2023-06-27 19:16:53.956 [info] > /bin/python3.11 ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/linter.py -m pydocstyle ./main.py
2023-06-27 19:16:53.956 [info] cwd: .
2023-06-27 19:16:53.996 [info] Discover tests for workspace name: test-vscode - uri: /home/user/temp/test-vscode/main.py
2023-06-27 19:16:54.060 [info] ##########Linting Output - pydocstyle##########

2023-06-27 19:16:54.060 [info] /home/user/temp/test-vscode/main.py:1 at module level:
        D100: Missing docstring in public module

2023-06-27 19:16:54.144 [info] ##########Linting Output - mypy##########

2023-06-27 19:16:54.144 [info] Success: no issues found in 1 source file

2023-06-27 19:17:20.080 [info] > ./.venv/bin/python -I ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/get_output_via_markers.py ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/interpreterInfo.py
2023-06-27 19:17:20.128 [info] Discover tests for workspace name: test-vscode - uri: /home/user/temp/test-vscode
2023-06-27 19:17:20.128 [info] Python interpreter path: ./.venv/bin/python
2023-06-27 19:17:20.282 [info] LINTING: Skipping linting from Python extension, since Flake8 extension is installed and enabled.
2023-06-27 19:17:20.309 [info] > ./.venv/bin/python ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/linter.py -m mypy --follow-imports=silent --ignore-missing-imports --show-column-numbers --no-pretty ./main.py
2023-06-27 19:17:20.309 [info] cwd: .
2023-06-27 19:17:20.334 [info] > ./.venv/bin/python ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/linter.py -m pydocstyle ./main.py
2023-06-27 19:17:20.335 [info] cwd: .
2023-06-27 19:17:20.383 [info] > ./.venv/bin/python -c "import mypy"
2023-06-27 19:17:20.408 [info] > ./.venv/bin/python -c "import pydocstyle"
2023-06-27 19:17:20.438 [info] > ./.venv/bin/python -c "import mypy"
2023-06-27 19:17:20.468 [info] > ./.venv/bin/python -c "import pydocstyle"
2023-06-27 19:17:20.471 [info] 
Linter 'mypy' is not installed. Please install it or select another linter".
Error: Module 'mypy' not installed.
2023-06-27 19:17:20.471 [warning] Linter 'mypy' is not installed. Please install it or select another linter". n [Error]: Module 'mypy' not installed.
    at s.result [as execForLinter] (/home/user/.vscode-server/extensions/ms-python.python-2023.10.1/src/client/common/process/pythonProcess.ts:47:23)
    at a.messages [as run] (/home/user/.vscode-server/extensions/ms-python.python-2023.10.1/src/client/linters/baseLinter.ts:98:19)
    at a.runLinter (/home/user/.vscode-server/extensions/ms-python.python-2023.10.1/src/client/linters/mypy.ts:20:15)
2023-06-27 19:17:20.495 [info] > . ./.venv/bin/activate && echo 'e8b39361-0157-4923-80e1-22d70d46dee6' && python ~/.vscode-server/extensions/ms-python.python-2023.10.1/pythonFiles/printEnvVariables.py
2023-06-27 19:17:20.495 [info] shell: bash
2023-06-27 19:17:20.498 [info] 
Linter 'pydocstyle' is not installed. Please install it or select another linter".
Error: Module 'pydocstyle' not installed.
2023-06-27 19:17:20.498 [warning] Linter 'pydocstyle' is not installed. Please install it or select another linter". n [Error]: Module 'pydocstyle' not installed.
    at s.result [as execForLinter] (/home/user/.vscode-server/extensions/ms-python.python-2023.10.1/src/client/common/process/pythonProcess.ts:47:23)
    at l.messages [as run] (/home/user/.vscode-server/extensions/ms-python.python-2023.10.1/src/client/linters/baseLinter.ts:98:19)
    at l.runLinter (/home/user/.vscode-server/extensions/ms-python.python-2023.10.1/src/client/linters/pydocstyle.ts:16:15)
2023-06-27 19:17:20.592 [warning] Retry failed, storage update failed for key PYTHON_ENV_INFO_CACHE
2023-06-27 19:17:20.592 [warning] Retry failed, storage update failed for key PYTHON_ENV_INFO_CACHE
2023-06-27 19:17:20.592 [warning] Retry failed, storage update failed for key PYTHON_ENV_INFO_CACHE
2023-06-27 19:17:20.592 [warning] Retry failed, storage update failed for key PYTHON_ENV_INFO_CACHE
2023-06-27 19:17:20.592 [warning] Retry failed, storage update failed for key PYTHON_ENV_INFO_CACHE
2023-06-27 19:17:20.592 [warning] Retry failed, storage update failed for key PYTHON_ENV_INFO_CACHE
2023-06-27 19:17:20.592 [warning] Retry failed, storage update failed for key PYTHON_ENV_INFO_CACHE
2023-06-27 19:17:20.592 [warning] Retry failed, storage update failed for key PYTHON_ENV_INFO_CACHE
2023-06-27 19:17:20.592 [warning] Retry failed, storage update failed for key PYTHON_ENV_INFO_CACHE
2023-06-27 19:17:20.592 [warning] Retry failed, storage update failed for key PYTHON_ENV_INFO_CACHE
2023-06-27 19:17:20.592 [warning] Retry failed, storage update failed for key PYTHON_ENV_INFO_CACHE
2023-06-27 19:17:20.592 [warning] Retry failed, storage update failed for key PYTHON_ENV_INFO_CACHE
2023-06-27 19:17:20.592 [warning] Retry failed, storage update failed for key PYTHON_ENV_INFO_CACHE
2023-06-27 19:17:20.592 [warning] Retry failed, storage update failed for key PYTHON_ENV_INFO_CACHE
2023-06-27 19:17:20.592 [warning] Retry failed, storage update failed for key PYTHON_ENV_INFO_CACHE
2023-06-27 19:17:20.592 [warning] Retry failed, storage update failed for key PYTHON_ENV_INFO_CACHE
2023-06-27 19:17:20.592 [warning] Retry failed, storage update failed for key PYTHON_ENV_INFO_CACHE
2023-06-27 19:17:20.592 [warning] Retry failed, storage update failed for key PYTHON_ENV_INFO_CACHE
2023-06-27 19:17:20.592 [warning] Retry failed, storage update failed for key PYTHON_ENV_INFO_CACHE
2023-06-27 19:17:20.592 [warning] Retry failed, storage update failed for key PYTHON_ENV_INFO_CACHE
2023-06-27 19:17:22.493 [info] > ./.venv/bin/python -c "import pip"
2023-06-27 19:17:22.550 [info] > ./.venv/bin/python -c "import pydocstyle"
2023-06-27 19:17:22.790 [info] Send text to terminal: source /home/user/temp/test-vscode/.venv/bin/activate
2023-06-27 19:17:23.290 [info] Send text to terminal: /home/user/temp/test-vscode/.venv/bin/python -m pip install -U pydocstyle
2023-06-27 19:17:23.394 [info] > ./.venv/bin/python -c "import pip"
2023-06-27 19:17:23.412 [info] Send text to terminal: /home/user/temp/test-vscode/.venv/bin/python -m pip install -U mypy
2023-06-27 19:17:23.442 [info] > ./.venv/bin/python -c "import mypy"
2023-06-27 19:17:31.471 [info] Send text to terminal: source /home/user/temp/test-vscode/.venv/bin/activate

```
Python Lang Server

```
2023-06-27 19:16:22.332 [info] [Info  - 7:16:22 PM] (1141065) Pylance language server 4.17.21 (pyright ae795f5b) starting
2023-06-27 19:16:22.333 [info] [Info  - 7:16:22 PM] (1141065) Server root directory: /home/user/.vscode-server/extensions/ms-python.vscode-pylance-2023.6.30/dist
2023-06-27 19:16:22.337 [info] [Info  - 7:16:22 PM] (1141065) Starting service instance "test-vscode"
2023-06-27 19:16:22.339 [info] [Info  - 7:16:22 PM] (1141065) Auto format for strings enabled
2023-06-27 19:16:22.469 [info] [Info  - 7:16:22 PM] (1141065) Found 1 source file
2023-06-27 19:17:20.263 [info] [Info  - 7:17:20 PM] (1141065) Found 1 source file
```

Terminal commands:
```
python3.10 -m venv .venv
source ./.venv/bin/activate
pip install fastapi

>(.venv) use@hostname:~/temp/test-vscode$ pip list | grep fastapi
fastapi           0.98.0
```

**Comment by mironnn** [comment]
<img width="944" alt="image" src="https://github.com/microsoft/pylance-release/assets/9884126/eca33f93-6b53-4e20-90ad-455a1fdb7c29">
<img width="497" alt="image" src="https://github.com/microsoft/pylance-release/assets/9884126/a23fcaf2-36b0-49ce-8cb4-640fdc63d05f">

**Comment by mironnn** [comment]
The package is installed in venv.


<img width="690" alt="Screenshot 2023-06-27 at 15 29 41" src="https://github.com/microsoft/pylance-release/assets/9884126/767b619b-ed8d-4698-a396-217f2b336fe8">

**Comment by heejaechang** [comment]
@mironnn not related to the issue, but just curios. where did you get the pylance extension? I am asking since the log says this

`Pylance language server 4.17.21 (pyright ae795f5b) starting` rather than ` Pylance language server 2023.6.30 (pyright ae795f5b) starting` which should be what normal pylance should print out.

**Comment by mironnn** [comment]
@heejaechang  Probably from the official extension.
> 

https://marketplace.visualstudio.com/items?itemName=ms-python.vscode-pylance

**Comment by mironnn** [comment]
@heejaechang Thank you for your question.

So I switched `Pylance` to the "Pre-release" version and everything is working as expected (v2023.6.31). If I return back to release v2023.6.30 - the issue still occurs.

And I still have the log with `2023-06-27 21:45:54.846 [info] [Info  - 9:45:54 PM] (1220217) Pylance (sync) language server 4.17.21 (pyright 3f84540e) starting`

What is changed:
using v2023.6.31 (extension installed)-> in the Python lang log: 4.17.21 (pyright ae795f5b) -> imports are not working
using v2023.6.30 (extension installed) -> in the Python lang log: 4.17.21 (pyright 3f84540e) -> import working


Full lof Python lang server with Pre-release Pylance extension v2023.6.31 :
```
2023-06-27 21:55:08.820 [info] [Info  - 9:55:08 PM] (1225691) Pylance (sync) language server 4.17.21 (pyright 3f84540e) starting
2023-06-27 21:55:08.820 [info] [Info  - 9:55:08 PM] (1225691) Server root directory: /home/user/.vscode-server/extensions/ms-python.vscode-pylance-2023.6.31/dist
2023-06-27 21:55:08.827 [info] [Info  - 9:55:08 PM] (1225691) Starting service instance "test-vscode"
2023-06-27 21:55:08.830 [info] [Info  - 9:55:08 PM] (1225691) Auto format for strings enabled
2023-06-27 21:55:08.876 [info] [Info  - 9:55:08 PM] (1225691) Setting pythonPath for service "test-vscode": "/home/user/temp/test-vscode/.venv/bin/python"
2023-06-27 21:55:08.879 [info] [Warn  - 9:55:08 PM] (1225691) stubPath /home/user/temp/test-vscode/types is not a valid directory.
2023-06-27 21:55:08.909 [info] [Info  - 9:55:08 PM] (1225691) Assuming Python version 3.10
2023-06-27 21:55:08.976 [info] [Info  - 9:55:08 PM] (1225691) Found 1 source file

```

**Comment by heejaechang** [comment]
just to verify. do you have this option `"python.analysis.enableSyncServer": true` enabled somewhere in your `setting.json` ? either `workspace`, `remote`, `user`, `profile` ?

if not, can you explicitly disable it `"python.analysis.enableSyncServer": false` in one of the settings you use and see whether it still repro.

**Comment by mironnn** [comment]
@heejaechang Yes, I have explicitly enabled `python.analysis.enableSyncServer`.
If I disable everything works as expected for the Pylance release and pre-release version.
Versions in the python lang server log are also correct.

So somewhere under my account there is out of date version? or how can I cleanup this?

**Comment by fsufitch** [comment]
+1

I am seeing the same issue as @mironnn , and it was also explicitly fixed by disabling the sync server. That's a workaround for now, I suppose. Thanks for that.

**Comment by heejaechang** [comment]
@mironnn can I ask why you enabled the setting?

@fsufitch are you saying you also explicitly enabled it? or it enabled automatically somehow?

the flag is `internal flag` for us to test some new internal changes and not supposed to be changed by users and that's why there is no explanation in `README`. 

Not just this flag but any other internal flags, if users touch them, there is no guarantee that all features will work as expected. we might need to put more strong wording in the description so people don't just blindly mess with them.

**Comment by mironnn** [comment]
@heejaechang To be honest, I'm just started using vscode. Don't know if I explicitly enabled it or if it appeared with some extension, or settings configuration,...
So I can disable it and everything should work as expected?

**Comment by fsufitch** [comment]
@heejaechang I explicitly enabled it, to tinker with the "Go to definition in string literal" feature (which I believe at some point said it required the sync server). I'm fine "beta testing" this kind of stuff, so I figured if/when it broke, I'd come to report it to you guys to help with development... and here I am.

These are the versions where I saw the issue, for reference:

* Pylance = `v2023.6.30`
* Python = `v2023.10.1`
* Pyright = `1.1.316` (in my venv, if that matters)

Again, my issue was resolved by disabling the sync server, so I'm not blocked on anything here -- especially since it's an experimental feature which is apparently not guaranteed to not break everything.
